cmd/proxy/main.go

```diff
@@ -40,31 +40,38 @@ func parseBackupURL(backupUrl string) (string, string, error) {
 	return u.String(), path, nil
 }
 
+// loggedError logs an error locally and returns the error decorated with the
+// log message so that it can be returned to the protobuf client.
+func loggedError(log logr.Logger, err error, message string) error {
+	log.Error(err, message)
+	return fmt.Errorf("%s: %s", message, err)
+}
```

Comment:
These errors were being returned direct to the restore-agent which was then panicing with an error which wasn't clear whether it came from the restore-agent code or from the proxy code.
I've added logging of errors to the proxy and decorate each error with some context.


```

ubectl -n teste2e-parallel-restore logs restored-cluster-0-restore 
Using etcd peer name: restored-cluster-0
Using etcd cluster name: restored-cluster
Using etcd initial cluster: restored-cluster-0=http://restored-cluster-0.restored-cluster.teste2e-parallel-restore.svc:2380,restored-cluster-1=http://restored-cluster-1.restored-cluster.teste2e-parallel-restore.svc:2380,restored-cluster-2=http://restored-cluster-2.restored-cluster.teste2e-parallel-restore.svc:2380
Using etcd advertise URL: http://restored-cluster-0.restored-cluster.teste2e-parallel-restore.svc:2380
Using etcd data directory: /var/lib/etcd
Using snapshot directory: /tmp/snapshot
Using proxy URL: eco-proxy.eco-system.svc:80
Using backup URL: s3://backups.test.improbable.io/snapshot.db?endpoint=minio.minio.svc&disableSSL=true&s3ForcePathStyle=true
Using timeout: 300 seconds
Downloading backup file from Proxy s3://backups.test.improbable.io/snapshot.db?endpoint=minio.minio.svc&disableSSL=true&s3ForcePathStyle=true
panic: rpc error: code = Unavailable desc = transport is closing

goroutine 1 [running]:
main.main()
	/workspace/cmd/restore-agent/main.go:103 +0x159f

```